### DISCUSSION TITLE
Multi Region

### DISCUSSION THREAD

**Discussion by Fezec** [Dynamoid/dynamoid#509] [Q&A]
I saw that in 2019 there was a request to specify region within the model. Given a Global Table use of dynamo, is it possible to change the service endpoint on the fly?  From us-east-1 to us-east-2 if an region is degraded?

**Comment by andrykonchin** [comment]
Right now you cannot change connection-related configuration (AWS credentials and region) on the fly in Dynamoid.  So it requires either restarting the application or monkey-patching to re-create AWS SDK client with a new config options.

I haven't considered this case with Global Table and switching to another region before. It isn't difficult to implement Dynamoid reconfiguration on the fly. But I have no idea how to detect degradation of DynamoDB table and what error DynamoDB will return in response (to trigger automatically switching to another region).

I will add this feature request (to allow reconfigure Dynamoid on the fly) to the TODO list and hope it will be released in the next version.

**Reply by Fezec** [reply]
if you can allow it, it can be left to the dev to know when to change it.  There are alerts in DynamoDb that are available such as replication latency which accounts for region degradation. 

That being said, is it possible to initialize 2 different models 1 using regionN and the other using another ? I tried it in a quick test but had no luck. Using a feature toggle solution, i was hoping to be able to swich regions to accomodate high availability and uptime.

**Reply by Fezec** [reply]
PS the 2019 post to which i refer is the following: https://github.com/Dynamoid/dynamoid/issues/403

**Reply by andrykonchin** [reply]
> That being said, is it possible to initialize 2 different models 1 using regionN and the other using another ?

It's still not implemented but I planned to release it in the next version (in June).

**Reply by Fezec** [reply]
That's great news. Do you have a specific Day? Is there a branch that may be available for pre-release?

**Reply by andrykonchin** [reply]
There is no strict schedule. Dynamoid is released every 3-4 months usually. The next version should have been released in this May but looks like will be postponed a bit. I hope it will be released in June.

All the changes are being merged into the `master` branch and it will be great to receive some feedback before the actual release. I will update all the issues related to the configuration improvement when it becomes available in `master`.

**Reply by Fezec** [reply]
Hello Andry, 
I just wanted to follow up and see if this feature has been released?

**Comment by andrykonchin** [comment]
Hello. Not yet, but it's the first item in the TODO list so I hope it will be done in the next several weeks.

**Comment by Fezec** [comment]
Hey Andry,
I light of events with AWS yesterday. I wanted to follow back up on this thread. Is there a way to change the region on the fly which the model refers yet?

**Reply by andrykonchin** [reply]
Hi. Unfortunately, I didn't have a chance to work on it :pensive:.
